Makefile

```diff
@@ -234,8 +235,18 @@ worker-key-pair: ca
 	rm /etc/osbuild-composer/worker-csr.pem
 
 .PHONY: unit-tests
+.ONESHELL:
 unit-tests:
-	go test -race ./...
+	go test -race -covermode=atomic -coverprofile=coverage.txt -coverpkg=$$(go list ./... | grep -v rpmmd/test$ | tr "\n" ",") ./...
+	# sub module is not tested automatically
```

Comment:
When I read this comment I was confused for a second (okay, not just a second, a few more) and thought this was about "git" submodules. Maybe `# go modules with go.mod in subdirs are not tested automatically` or something like this?

But this makes me wonder if instead of adding special cases here we could just move `splunk_logger` into it's own git repo (e.g. `github.com/osbuild/splunk_logger` ) and then use the normal vendor mechanism to get it (just like any other dep). Then the tests for spluk_logger would run in the splunk_logger git repo (just like any other dep). Wdyt? Maybe too big of a change for such a tiny package but it would feel slightly cleaner to me (but maybe I'm missing something here) (adding @achilleas-k as it seems he and Diaa did some work on this)

Reply:
for sure I'll take your suggestion as new comment, thanks!

Totally agree, this should be separate. Also @croissanne emphasized several times, this should be split
…or even some library containing more than `splunk_logger`?
but for now the tests should at least run - if I mess with them ;)

Reply:
1. Call it a subpackage (Go calls them packages) to avoid confusion.
2. Let's split it out in a follow-up.

Reply:
If desired I can do the split tomorrow

Reply:
Merging this and splitting sounds good to me. Took the liberty of creating https://github.com/osbuild/splunk_logger  :)